.github/workflows/check_formatting.yaml

```diff
@@ -18,71 +18,22 @@ name: Check Formatting
 
 on:
   workflow_call:
-    inputs:
-      license_config:
-        description: "The license configuration file for SkyWalking Eyes"
-        type: string
-        required: false
-        default: ".licenserc.yaml"
-      cpp_source_dirs:
-        description: "Space seperated list of dirs to apply clang-format to"
-        type: string
-        required: false
-        default: "include src tests"
-      format_python:
-        description: "Whether or not to format python files"
-        type: boolean
-        required: false
-        default: true
 
 jobs:
-  # Check licensing.
-  check_licensing:
-    if: inputs.license_config != ''
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout Source
-        uses: actions/checkout@v4
-      - name: Check for License Config
-        if: hashFiles(inputs.license_config) == ''
-        uses: actions/github-script@v3
-        with:
-          script: |
-            core.setFailed("License Config file doesn't exist: ${{inputs.license_config}}")
-      - name: Check License
-        uses: apache/skywalking-eyes@v0.4.0
-        with:
-          config: ${{ inputs.license_config }}
-          mode: check
-
-  # Check C++ formatting.
-  check_cpp_format:
-    if: inputs.cpp_source_dirs != ''
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout Source
-        uses: actions/checkout@v4
-      - name: Check C++ Formatting
-        uses: DoozyX/clang-format-lint-action@v0.20
-        with:
-          source: ${{ inputs.cpp_source_dirs }}
-          extensions: "hpp,cpp,ipp,h,c"
-          clangFormatVersion: 20
-          inplace: False
-
-  # Check Python formatting.
-  check_python_format:
-    if: inputs.format_python == true
+  pre-commit:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Source
         uses: actions/checkout@v4
+      - name: Download .pre-commit-config.yaml
+        run: wget https://raw.githubusercontent.com/NWChemEx/pre-commit-config/master/.pre-commit-config.yaml
+        shell: bash
       - uses: actions/setup-python@v2
         with:
           python-version: "3.x"
-      - name: Install black
-        run: pip install black
+      - name: Install pre-commit
+        run: pip install pre-commit
```

Comment:
We had issues with incorrect versions of certain packages being picked up on a CI runner in CMakePP/CMinx#165. I'd recommend creating and using a fresh virtual environment each run to help avoid potential issues here. Take a look at [CMinx/.github/workflows/test_pip_install.yaml](https://github.com/CMakePP/CMinx/blob/master/.github/workflows/test_pip_install.yaml) to see how we did it there, using `$GITHUB_OUTPUT` to pass the venv information between steps.